### PR TITLE
Query formatting and dependency tree/graph visualisation

### LINKED ISSUE #804: print_dependency_tree() should allow showing info whether queries are stored
It would be useful if `print_dependency_tree()` had an option to show whether the dependent queries have been stored or not.

### PR DESCRIPTION
**Pull request by maxalbert**
Closes #804.
Related to #786.

### I have:

- [X] Formatted any Python files with [black](https://github.com/ambv/black)
- [ ] Brought the branch up to date with master
- [X] Added any relevant Github labels
- [X] Added tests for any new additions
- [ ] Added or updated any relevant documentation
- [ ] Added an Architectural Decision Record (ADR), if appropriate
- [ ] Added an [MPLv2 License Header](https://www.mozilla.org/en-US/MPL/headers/) if appropriate
- [X] Updated the [Changelog](https://github.com/Flowminder/FlowKit/blob/master/CHANGELOG.md) 

### Description

This PR adds an optional argument `show_stored` to `print_dependency_tree()` which allows to display information on whether dependent queries have been stored:
```python
>>> dl = daily_location(date="2016-01-01", method="last")
>>> print_dependency_tree(dl, show_stored=True)
<Query of type: LastLocation, query_id: 'b46b86f42634affb00c510a1778134d4', is_stored: False>
  - <Query of type: JoinToLocation, query_id: '89a463385d85be2e77e5003b3f3b0403', is_stored: False>
     - <Query of type: _SubscriberCells, query_id: '8c568b3b371ecbb943d7a1ad35f1d1b0', is_stored: False>
        - <Query of type: EventsTablesUnion, query_id: '01f9cca502930c3a7322f00121ef9dec', is_stored: False>
           - <Query of type: EventTableSubset, query_id: 'd3b49c25d7b2ec64693e75fdae222fd9', is_stored: False>
              - <Query of type: Table, query_id: '9de507e882f1fb6b0cfcedd324e27839', is_stored: True>
                 - <Query of type: Table, query_id: '7a7f27978925c385bc44a5ec5667d7b3', is_stored: True>
           - <Query of type: EventTableSubset, query_id: 'd5cf1feacd710753f51b630b59c2deee', is_stored: False>
              - <Query of type: Table, query_id: '70a2d392f73aedcc251a6a504f17008a', is_stored: True>
                 - <Query of type: Table, query_id: '057addedac04dbeb1dcbbb6b524b43f0', is_stored: True>
     - <Query of type: CellToAdmin, query_id: 'f73369cb975cfd8c13a50f9ab1754106', is_stored: False>
        - <Query of type: CellToPolygon, query_id: 'e0196a13e5bf8d7c0c62fe5a6c63f87e', is_stored: False>
 ```

To implement this I also added a `Query.__format__` method which allows more convenient formatting of query objects by optionally specifying attributes that should be included in the resulting string, for example:
```python
>>> dl = daily_location(date="2016-01-01", method="last")
>>> print(f"{dl}")
<Query of type: LastLocation>
>>> print(f"{dl:is_stored,query_state}")
<Query of type: LastLocation, is_stored: False, query_state: <QueryState.KNOWN: 'known'>>
```

Finally, since I already needed to copy & paste the example code from #786 a few times, I decided to wrap it up in a helper function `plot_dependency_tree()` which can plot a dependency graph in SVG or PNG format, so that it can be conveniently displayed in a Jupyter notebook.

### CODE REVIEW

**Comment by maxalbert** [comment]
Hmm, there is some alleged drop in coverage but I think this is due to codecov not picking up `test_utils.py` (#789)?

File: flowmachine/tests/test_query.py [hunk 1]
```diff
@@ -135,3 +135,20 @@ def test_make_sql_no_overwrite():
 
     dl = daily_location("2016-01-01")
     assert [] == dl._make_sql("admin3", schema="geography")
+
+
+def test_query_formatting():
+    """
+    Test that query can be formatted as a string, with optional
```
Comment (jch1g10) [inline flowmachine/tests/test_query.py]: With optional...?

File: flowmachine/flowmachine/core/query.py [hunk 2]
```diff
@@ -112,16 +112,51 @@ def md5(self):
...
     @abstractmethod
     def _make_query(self):
 
         raise NotImplementedError
 
     def __repr__(self):
+        # Default representation, derived classes might want to add something more specific
+        return format(self, "query_id")
+
+    def __format__(self, fmt=""):
```
Comment (greenape) [inline flowmachine/flowmachine/core/query.py]: Very neat :)

File: flowmachine/tests/test_query.py (hunk 1 above)
Comment (maxalbert) [reply flowmachine/tests/test_query.py]: Thanks, fixed.

**Pull request merged**